src/s390x/zipl.rs

```diff
@@ -58,14 +59,18 @@ pub fn install_bootloader<P: AsRef<Path>>(
         .arg("--target")
         .arg(boot.as_ref().as_os_str())
         .status()
-        .chain_err(|| "failed to execute zipl")?;
-    error_chain::ensure!(status.success(), "installing bootloader");
+        .chain_err(|| format!("failed to execute zipl on {}", disk))?;
+    if !status.success() {
+        bail!("couldn't install bootloader on {}", disk);
+    }
 
     let status = Command::new("chreipl")
         .arg(disk)
         .status()
-        .chain_err(|| "failed to execute chreipl")?;
-    error_chain::ensure!(status.success(), "setting {} as boot-device", disk);
+        .chain_err(|| format!("failed to execute chreipl on {}", disk))?;
+    if !status.success() {
```

Comment:
is there any reason to use `if - bail` instead of `ensure` ?

Reply:
IMO it's better to make control flow explicit, rather than hiding it in a macro argument.